api/v1alpha3/types.go

```diff
@@ -327,18 +333,46 @@ type SubnetSpec struct {
 	Role SubnetRole `json:"role,omitempty"`
 
 	// ID defines a unique identifier to reference this resource.
+	// +optional
 	ID string `json:"id,omitempty"`
 
 	// Name defines a name for the subnet resource.
 	Name string `json:"name"`
 
 	// CidrBlock is the CIDR block to be used when the provider creates a managed Vnet.
+	// +optional
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
 	// InternalLBIPAddress is the IP address that will be used as the internal LB private IP.
 	// For the control plane subnet only.
+	// +optional
 	InternalLBIPAddress string `json:"internalLBIPAddress,omitempty"`
 
 	// SecurityGroup defines the NSG (network security group) that should be attached to this subnet.
+	// +optional
 	SecurityGroup SecurityGroup `json:"securityGroup,omitempty"`
+
+	// RouteTable defines the route table that should be attached to this subnet.
+	// +optional
+	RouteTable RouteTable `json:"routeTable,omitempty"`
+}
+
+// GetControlPlaneSubnet returns the cluster control plane subnet.
+func (n *NetworkSpec) GetControlPlaneSubnet() *SubnetSpec {
```

Comment:
Just a comment, but it seems a bit odd and possibly error prone to use a list and expect only one of role to be present selecting the first item with that role.

Reply:
yep, totally...

Trying to capture that in https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/618

Reply:
I think right now the code makes the false assumption that there is one subnet per role (and will only provision one subnet per role) so I think we need to change it to allow multiple node subnets, and not store that subnets in a slice of pointers...

Reply:
💯

Reply:
also filed https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/664 to talk about supporting multiple subnets specifically, thanks for the nudge